src/main/java/io/oneko/project/rest/ProjectController.java

```diff
@@ -100,6 +102,15 @@ ProjectDTO getProjectById(@PathVariable UUID id) {
 				.orElseThrow(() -> new ResponseStatusException(HttpStatus.NOT_FOUND, "Project with id " + id + " not found"));
 	}
 
+	@PreAuthorize("hasAnyRole('ADMIN', 'DOER', 'VIEWER')")
+	@GetMapping("/byDeploymentUrl")
+	ProjectDTO getProjectByDeploymentUrl(@RequestParam String deploymentUrl) {
```

Comment:
I'm not sure whether or not this is generally considered bad practice, but at least I personally don't like having URLs as URL-Parameters in other URLs, since it is making the whole thing somewhat hard to read.
I could see the advantage of using a Post-Request here to throw the URL into a reqeust body. I leave this up to you though, since this naturally is more like a get-request.

Reply:
I am somewhat surprised, that this returns a ProjectDTO instead of a ProjectVersionDTO. A single URL belongs to a specific version rather then a project. Of course in that case this couldn't be a regular versionDTO, since we probably want to include some information regarding the parent project as well.

Reply:
I realized we don't need this method anymore because Catnip already has access to all URLs and can look it up without an API call (after I changed some bits in Catnip).